### ISSUE TITLE
savefig.transparent has no effect when saving from UI

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#9080]
### Bug report

**Bug summary**

The savefig.transparent rcparam does not affect figures saved from the GUI.

**Code for reproduction**

```python
from pylab import *
rcParams["savefig.transparent"] = True
plot()
show()
```
then click on the save button to save a figure.

**Actual outcome**

The figure has opaque background.

**Expected outcome**

The figure should have transparent background, just as it has when calling savefig().

**Matplotlib version**

  * Operating System:
  * Matplotlib Version: 2.0.2 but apparently dates back to at least 1.5.0.
  * Python Version:
  * Jupyter Version (if applicable):
  * Other Libraries:

**Comment by WeatherGod** [comment]
Didn't we just have an issue go by about figure background transparency?

**Comment by anntzer** [comment]
I think this is different: I am specifically talking about a difference in behavior between interactive save and savefig.

**Comment by tacaswell** [comment]
I suspect this has been here much longer and is related to the some what convoluted method that `print_figure` is finally called either going through the 'save' button or `savefig`.  I have a vague memory of trying to trace through all of that a few years ago and giving up.

**Comment by u55** [comment]
There was a related pull request here: #6197 for `savefig.facecolor` and `savefig.edgecolor`. It should be simple to add `savefig.transparent`.

**Comment by anntzer** [comment]
Looks like the following chunk from Figure.savefig
```
        if transparent:
            kwargs.setdefault('facecolor', 'none')
            kwargs.setdefault('edgecolor', 'none')
            original_axes_colors = []
            for ax in self.axes:
                patch = ax.patch
                original_axes_colors.append((patch.get_facecolor(),
                                             patch.get_edgecolor()))
                patch.set_facecolor('none')
                patch.set_edgecolor('none')
```
should be moved to print_figure (which savefig ultimately calls, so there's no need to keep it in both places).

**Comment by tacaswell** [comment]
Sounds reasonable.

The restore code would also have to move.

**Comment by anntzer** [comment]
I think the correct strategy is
- quickfix: use savefig everywhere instead of print_figure,
- later: move everything from savefig to print_figure, so that savefig is a completely empty wrapper for print_figure (we could also have moved everything from print_figure to savefig and have print_figure be a minimal dispatcher to the print_* methods, but that would cause a backcompatibility break as print_figure would lose some capabilities).

**Comment by anntzer** [comment]
Closed by #9124.
